app/Main.hs

```diff
@@ -133,11 +133,11 @@ eventHandler = \case
         :: MonadIO m
         => Text
         -> Next (VgrepT AppState m Redraw)
-    handleFeedResult line = Continue $ do
-        expandedLine <- expandLineForDisplay line
-        case parseLine expandedLine of
-            Just l  -> zoom results (feedResult l)
-            Nothing -> pure Unchanged
+    handleFeedResult line = Continue $ case parseLine line of
+        Just l  -> do
+            l' <- traverseOf (lineReference . lineText) expandFormattedLine l
+            zoom results (feedResult l')
```

Comment:
I think `id` is *much* less easy to read. Just imagine you come along this piece of code and don’t know it’s from Lens! And if that’s unconvincing, remember that `($) = id` as well, up to a fixity declaration.